mobile/android/app/src/main/kotlin/app/alextran/immich/widget/model/Model.kt

```diff
@@ -49,6 +49,20 @@ enum class WidgetConfigState {
   LOADING, SUCCESS, LOG_IN, NO_CONNECTION
 }
 
+enum class WidgetImageFilter {
+  NONE, BLACK_WHITE;
+  
+  companion object {
+    fun fromString(value: String?): WidgetImageFilter {
+      return try {
+        valueOf(value ?: NONE.name)
+      } catch (e: IllegalArgumentException) {
+        NONE
+      }
+    }
+  }
```

Comment:
Lets make an extension of WidgetImageFilter
```suggestion
fun WidgetImageFilter.apply(src: Bitmap): Bitmap =
    when (this) {
        WidgetImageFilter.NONE -> src
        WidgetImageFilter.BLACK_WHITE -> applyBlackAndWhiteFilter(src)
    }
```

so we can call it like filter.apply(bitmap)